packages/@sanity/default-layout/src/components/Search.js

```diff
@@ -67,11 +67,12 @@ function search(query) {
       )
   )
 
-  const terms = query.split(/\s+/)
+  const terms = query.split(/\s+/).filter(item => item !== '')
   const uniqueFields = union(searchableFields)
-  const constraints = flatten(uniqueFields.map(field => terms.map(term => `${field} match '${term}**'`)))
-
-  return client.observable.fetch(`*[${constraints.join(' || ')}][0...10]`)
+  const constraints = terms.map(term => uniqueFields.map(field => `${field} match '${term}*'`))
+  const constraintString = constraints.map(constraint => `(${constraint.join(' || ')})`).join(' && ')
+  const searchQuery = `*[${constraintString}][0...10]`
```

Comment:
Minor nitpick: You could skip this variable and just pass the string to .fetch below (kind of obvious that it is the search query, so the extra variable feels a bit redundant IMO)